src/Microsoft.AspNetCore.Mvc.Modules/Extensions/ServiceCollectionBuilderExtensions.cs

```diff
@@ -67,6 +69,7 @@ public static IMvcCoreBuilder AddExtensionsApplicationParts(this IMvcCoreBuilder
             var availableExtensions = extensionManager.GetExtensions();
             using (logger.BeginScope("Loading extensions"))
             {
+                ConcurrentBag<Assembly> bagOfAssemblies = new ConcurrentBag<Assembly>();
                 Parallel.ForEach(availableExtensions, new ParallelOptions { MaxDegreeOfParallelism = 4 }, ae =>
```

Comment:
Not sure in this context but if we still load extensions as we currently do. In the loop `LoadExtensionAsync()` is called which triggers dyn compilation and / or loading.